### PR TITLE
refactor: simplify notifiers

### PR DESCRIPTION
**Pull request by iProdigy**
* Simplify `BaseNotifier` and add `#isEnabled` for improved structure in subclasses (this is also in the spirit of the upstream refactor, but slightly different)
* Remove `client` state in `Utils`
* Simplify the logic in `DiscordMessageHandler`
* Optimize `Utils.reduceItemStack` from $O(n \cdot m^2)$ to $O(n)$
* Remove unused `WorldService`
* Optimize config string replacements (avoid regex)
* Remove `isIgnoredWorld` from `DinkPlugin`
* Skip pet notifications in ignored worlds

### CODE REVIEW

**Review by pajlada — CHANGES_REQUESTED** [review]
Small nitpicks

Code overall looks good, although the diff was hell to go through :imp: 

I think these sort of overarching changes should take place as separate PRs in the future, so for example:
PR 1: Remove `client` state in `Utils`
PR 2: Simplify `BasENotifier` and add `#isEnabled` for improved structure in subclasses
PR 3: Skip pet notifications in ignored worlds
PR 4: Simplify logic in `DiscordMessageHandler`
PR 5: Optimize config string replacements
PR 6: Move around functions in notifier classes so they're consistently placed

File: src/main/java/dinkplugin/notifiers/ClueNotifier.java [hunk 1]
```diff
@@ -34,108 +36,100 @@ public ClueNotifier(DinkPlugin plugin) {
...
+            SerializedItemStack stack = new SerializedItemStack(itemId, quantity, price, itemComposition.getName());
+
+            totalPrice.addAndGet(stack.getTotalPrice());
+            itemStacks.add(stack);
+            lootMessage.append(getItemMessage(stack, embeds));
+        });
 
-        if (totalPrice < plugin.getConfig().clueMinValue()) {
+        if (totalPrice.get() < plugin.getConfig().clueMinValue())
             return;
```
[the remark this answers is not in the document]
Comment (iProdigy) [reply src/main/java/dinkplugin/notifiers/ClueNotifier.java:106]: Yes, good catch!

File: src/main/java/dinkplugin/Utils.java [hunk 2]
```diff
@@ -41,26 +40,9 @@ public static byte[] convertImageToByteArray(BufferedImage bufferedImage) throws
...
-                list.add(new ItemStack(item.getId(), item.getQuantity() + quantity, item.getLocation()));
-            } else {
-                list.add(item);
-            }
-        }
-
-        return list;
+        final Map<Integer, ItemStack> itemById = new HashMap<>();
+        items.forEach(item -> itemById.merge(item.getId(), item, SUM_ITEM_QUANTITIES));
+        return itemById.values();
```
[the remark this answers is not in the document]
Comment (iProdigy) [reply src/main/java/dinkplugin/Utils.java:45]: Agree - happy to add the test to this PR if you don't mind a little more clutter

**Review by pajlada — APPROVED** [review]
Code looks good :+1: I'll do a quick test of some of the functionality and then merge this in

**Comment by pajlada** [comment]
I tested: levelup, loot, death, slayer, clue scrolls and the webhooks fired & the extra data looked correct to me

**Pull request merged**